### DISCUSSION TITLE
Dedicated Server - Implement Reading From stdin Redirect

### DISCUSSION THREAD

**Discussion by j-r-g-d** [FakeFishGames/Barotrauma#13069] [Ideas and suggestions]
Reading from stdin and executing as commands as an alternative to the interactive server console is a fairly essential feature for dedicated game servers to have for automation. It is evident that Barotrauma is detecting the presence of stdin redirection, but hasn't actually implemented reading and executing from it. To better automate and handle dedicated servers of the game, reading from redirected stdin should be implemented.

**Comment by cl0ck-byte** [comment]
Still a thing to this day and it's absolutely nuts that you can't even send a SIGINT/SIGTERM to shut down the server.

**Comment by Regalis11** [comment]
Thank you for the report! This has now (finally) been [addressed in our private repo](https://github.com/FakeFishGames/Barotrauma-development/pull/8320), and the change will be included in the next update.

> Adds the command line argument -simpleio that enables simple io mode for dedicated server, allowing them to properly work under simulated TTYs that don't support all the fancy console features (e.g. pterodactyl)
